src/main/java/com/amazon/opendistroforelasticsearch/security/auth/UserInjector.java

```diff
@@ -66,10 +66,34 @@
 
     }
 
-    boolean injectUser(RestRequest request) {
+    public static class InjectedUser extends User {
+        private InetAddress inetAddress;
+        private int port = -1;
+
+        public InjectedUser(String name) {
+            super(name);
+        }
+
+        public InetAddress getInetAddress() {
```

Comment:
Is it necessary to have separate getter and setter for `InetAddress` and port? Should it be sufficient to provide getter for `TransportAddress` and setter that takes `String` and parses it?

Reply:
Marked transport as transient for serialization.